price-tolerance/src/main.nr

```diff
@@ -1,9 +1,12 @@
 mod float;
-use float::{Float, lte, div, sub, abs, printlnF};
+use float::{Float, lte, div, sub, abs, printlnF, sum};
 
 global MAX_VALIDATORS = 8;
 
-fn main(validators: pub [Float; MAX_VALIDATORS], tolerance: pub Float) {
+fn main(
+    validators: pub [Float; MAX_VALIDATORS],
+    tolerance: pub Float
+) -> pub ([Float; MAX_VALIDATORS], u32, u32) {
```

Comment:
Do we need to return `n`, why would someone who wants to know which validators to award for their contribution require the total number of validators, including the wrong ones? It may even not be needed to return `correct_count`, we would either have a correct or a trivial value.

Finally, we should be returning the median! In case we have an even count of values, the median may not equal any value in our array, so we should just return it as a separate value.

Reply:
The median value is defined to be the average of the two inner-most values in case of an even number of elements, so this is correct as it is written. 

We're returning the number of validators that are in the committee and the number of valid reports, because the sequencer would like to know the percentage of correct values out of the total number of participants, to compute a threshold based upon which he will make a decision if a majority has been reached and to publish to our smart-contracts or to omit. 

It's true that you can get the `correct_count` through counting the number of validators and `n` by the size of the input value array, however it would be difficult to prove to a third party that this is indeed the correct computation and that no value has been maliciously omitted.

Reply:
> The median value is defined to be the average of the two inner-most values in case of an even number of elements, so this is correct as it is written.

I disagree, I'm certain we must return the actual median value.

What about the case of "1 2 3 50", assuming everything but 50 is within the threshold, then we would output "1 2 3", but how could we extract the median 2.5?

Or in a similar fashion, "1 20 30 40", where we would return "20 30 40", we still cannot find the median 25! The sense of "middle" is lost when we remove invalid values.

Even if we return all values and the number of correct ones (as we do now), the sequencer couldn't figure out which are correct without running the entire algorithm, since they could all be a bit less than the median, a bit more or both (again, we don't have a sense of positioning).

> the sequencer would like to know the percentage of correct values out of the total number of participants

Then I believe it makes much more sense to straight up return the percentage.